interop-cats/jvm/src/main/scala/scalaz/zio/interop/catszmanagedjvm.scala

```diff
@@ -0,0 +1,109 @@
+package scalaz.zio.interop
+
+import cats.effect.Resource.{ Allocate, Bind, Suspend }
+import cats.effect.{ ExitCase, LiftIO, Resource, Sync, IO => CIO }
+import cats.{ effect, Bifunctor, Monad, MonadError, Monoid, Semigroup, SemigroupK }
+import scalaz.zio.{ Reservation, ZIO, ZManaged }
+import scalaz.zio.Runtime
+
+trait CatsZManagedSyntax {
+
+  implicit class ResourceSyntax[A](resource: Resource[CIO, A]) {
+    def toManaged[R, E](
+      implicit l: LiftIO[ZIO[R, Throwable, ?]]
+    ): ZManaged[R, Throwable, A] = {
+      def convert[A1](resource: Resource[CIO, A1]): ZManaged[R, Throwable, A1] =
+        resource match {
+          case Allocate(res) =>
+            ZManaged.unwrap(
+              l.liftIO(res)
+                .map {
+                  case (a, r) =>
+                    ZManaged.reserve(Reservation(ZIO.succeed(a), l.liftIO(r(ExitCase.Completed)).orDie.uninterruptible))
+                }
+                .uninterruptible
+            )
+          case Bind(source, fs) =>
+            convert(source).flatMap(s => convert(fs(s)))
+          case Suspend(res) =>
+            ZManaged.unwrap(l.liftIO(res).map(convert))
+        }
+      convert(resource)
+    }
+  }
+
+  implicit class ZManagedSyntax[R, E, A](managed: ZManaged[R, E, A]) {
+    def toResource[F[_]](implicit r: Runtime[R], S: Sync[F]): Resource[F, A] =
+      Resource.suspend(S.delay {
+        r.unsafeRun(managed.reserve) match {
```

Comment:
Hmm why `unsafeRun` and not a conversion to `F`?

Reply:
This is inspired by how it's done in the STM interop
https://github.com/scalaz/scalaz-zio/blob/master/interop-cats/jvm/src/main/scala/scalaz/zio/interop/stm/STM.scala#L180-L185

The alternative would only work with a ZIO ~> F as far as I'm aware and John mentioned to prefer this approach

Reply:
👍